### DISCUSSION TITLE
Replacing `git2` with `gitoxide`

### DISCUSSION THREAD

**Discussion by blinxen** [sharkdp/bat#3151] [Ideas]
I recently took over the maintenance of the `bat` package in Fedora and was wondering if the maintainers would be willing to accept a PR that replaces the usage of `git2` with `gitoxide`. `git2` is a pain to deal with and since I am also maintaining the `gitoxide` stack in Fedora, it would ease the maintenance a lot .

**Comment by keith-hall** [comment]
I personally would be in favor of dropping the dependency on libgit2 and use a more performant Rust implementation 👍

**Reply by keith-hall** [reply]
Motivation: https://github.com/sharkdp/bat/issues/3029#issuecomment-2648540582

**Comment by sharkdp** [comment]
I'd also be okay with this.
